packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -15,8 +16,35 @@ const LOCATION_SEARCH_OPTIONS = {
 
 const LOCATION_SEARCH_CONTAINER = 'geocoder-container';
 
+interface LocationSearchProps extends MaplibreGeocoderOptions {
+  /**
+   * Emitted when the input is cleared
+   */
+  onClear: () => void;
+  /**
+   * Emitted when the geocoder is looking up a query
+   */
+  onLoading: (query) => void;
+  /**
+   * Fired when the geocoder returns a response
+   */
+  onResults: (results) => void;
+  /**
+   * Fired when input is set
+   */
+  onResult: (result) => void;
+  /**
+   * Emitted on error as string
+   */
+  onError: (error) => void;
+}
+
+type eventTypes = 'clear' | 'loading' | 'result' | 'results' | 'error';
```

Comment:
TypeScript uses PascalCase and prefers single vs plural naming for union types, suggest:
```suggestion
type EventType = 'clear' | 'loading' | 'result' | 'results' | 'error';
 ```